lib/elixir/lib/function.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Function do
+  @moduledoc """
+  A set of funtions for working with functions.
+
+  We can have two types of captures functions - external and local -
+  external functions are our normal functions residing in modules while
+  local are the anonymous functions defined with `fn` or the capture
+  operator `&`.
+  """
+
+  @doc """
+  Captures a function by a given module, function name and arity.
+
+  Useful when you want to capture a function when having variables.
+
+  ## Examples
+
+    iex> Function.capture(String, :length, 1)
+    &String.length/1
+  """
+  @type information ::
+          :type
+          | :module
+          | :arity
+          | :name
+          | :env
+          | :pid
+          | :index
+          | :new_index
+          | :new_uniq
+          | :uniq
+  @type value :: any
+
+  @spec capture(atom, atom, integer) :: (... -> any)
+  def capture(mod, fun, arity) when is_atom(mod) and is_atom(fun) and is_integer(arity) do
```

Comment:
I woulf rename args mod to module and fun to function_name.
Fun is used when they are anonimous functions, not atoms

Reply:
Guard for arity should be
is_integer(arity) and arity >= 0 and arity <= 255